pilot/pkg/bootstrap/server.go

```diff
@@ -655,7 +659,7 @@ func (s *Server) initEventHandlers() error {
 		configHandler := func(old, curr model.Config, _ model.Event) {
 			pushReq := &model.PushRequest{
 				Full:           true,
-				ConfigsUpdated: map[resource.GroupVersionKind]map[string]struct{}{curr.GroupVersionKind(): {}},
+				ConfigsUpdated: map[resource.GroupVersionKind]map[string]struct{}{curr.GroupVersionKind(): {curr.Name: {}}},
```

Comment:
Name is not unique, should we make this `name.namespace`?

Reply:
Yeah. I've considered that. But it's annoying to distinguish "." in name itself with that in "name.namespace". Thus I'd like to leave it for further improvement and mix-up the resource with same name but from difference namespace just not scope that well as that rarely happens.

Maybe later we can use a struct {Name, Namespace string} to describe a changed resource.

Reply:
I think its reasonable to not do it since its an optimization not a matter of correctness. We could make it something like `name~namespace` or similar though, just need the separator to not be a valid name?

Reply:
I agree with the point "an optimization not a matter of correctness". 
But if we involve the namespace, there're some more work to do such as recording name+namespace rather than only name in sidecarScope and the places sending PushRequest doing this fmt.Sprintf. It seems a little complex, do u think that's worthy?

Reply:
See comment below about making it a struct, I think that is simpler in general and solves this